Lib/pathlib.py

```diff
@@ -1177,6 +1178,19 @@ def replace(self, target):
         os.replace(self, target)
         return self.__class__(target)
 
+    def move(self, target, copy_function=shutil.copy2):
+        """
+        Recursively move a file or directory to another location (target),
+        using ``shutil.move``.
+
+        If *target* is on the current filesystem, then ``os.rename()`` is used.
+        Otherwise, *target* will be copied using *copy_function* and then removed.
+
+        Returns the new Path instance pointing to the target path.
+        """
+        shutil.move(self, target, copy_function)
+        return self.__class__(target)
```

Comment:
I think it would be better to return the actual destination of the file, i.e. the return value of `shutil.move()`,

Reply:
Would it not violate the API of all the module? All the methods here seems to wrap file paths with `Path`.

Reply:
Ah sorry, and wrap it in a Path:

```
return self.__class__(shutil.move(self, target, copy_function))
```

(this is _not_ the same as *target* - see the implementation of `move()`!)

Reply:
Done! Thanks for that sharp eye!